packages/e2e/lib/e2e_driver.dart

```diff
@@ -1,21 +1,89 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
 import 'dart:async';
+import 'dart:convert';
 import 'dart:io';
 
-import 'package:e2e/common.dart' as e2e;
 import 'package:flutter_driver/flutter_driver.dart';
 
-Future<void> main() async {
+import 'package:e2e/common.dart' as e2e;
+import 'package:path/path.dart' as path;
+
+/// This method remains for backword compatibility.
+Future<void> main() => e2eDriver();
+
+/// Flutter Driver test output directory.
+///
+/// Tests should write any output files to this directory. Defaults to the path
+/// set in the FLUTTER_TEST_OUTPUTS_DIR environment variable, or `build` if
+/// unset.
+String testOutputsDirectory =
+    Platform.environment['FLUTTER_TEST_OUTPUTS_DIR'] ?? 'build';
+
+/// The callback type to handle [e2e.Response.data] after the test succcess.
+typedef ResponseDataCallback = FutureOr<void> Function(Map<String, dynamic>);
+
+/// Writes a json-serializable json data to to
+/// [testOutputsDirectory]/`testOutputFilename.json`.
+///
+/// This is the default `responseDataCallback` in [e2eDriver].
+Future<void> writeResponseData(
+  Map<String, dynamic> data, {
+  String testOutputFilename = 'e2e_perf_summary',
+}) async {
+  assert(testOutputFilename != null);
+  await fs.directory(testOutputsDirectory).create(recursive: true);
+  final File file = fs.file(path.join(
+    testOutputsDirectory,
+    '$testOutputFilename.json',
+  ));
+  final String resultString = _encodeJson(data, true);
+  await file.writeAsString(resultString);
+}
+
+/// Adaptor to run E2E test using `flutter drive`.
+///
+/// `timeout` controls the longest time waited before the test ends.
+/// It is not necessarily the execution time for the test app: the test may
+/// finish sooner than the `timeout`.
+///
+/// `responseDataCallback` is the handler for processing [e2e.Response.data].
+/// The default value is `writeResponseData`.
+///
+/// To an E2E test `<test_name>.dart` using `flutter drive`, put a file named
+/// `<test_name>_test.dart` in the app's `test_driver` directory:
+///
+/// ```dart
+/// import 'dart:async';
+///
+/// import 'package:e2e/e2e_driver.dart' as e2e;
+///
+/// Future<void> main() async => e2e.e2eDriver();
+///
+/// ```
+Future<void> e2eDriver({
+  Duration timeout = const Duration(minutes: 1),
+  ResponseDataCallback responseDataCallback = writeResponseData,
```

Comment:
Default to write `response.data` into `e2e_perf_summary.json` seems a little strange as it doesn't necessarily have perf summary data. Shall we default to not write any file, and provide a `perfResponseCallback` that writes `response.data['performance']` into `e2e_perf_summary.json`? That seems a little more helpful in updating your current code in https://github.com/flutter/flutter/pull/61509/files.

Reply:
I probably should change the file name to some more general ones. Getting `performance` entry is just one more line like in https://github.com/flutter/flutter/pull/62064/files#diff-73e66281b14c792afcdcd20acb9edd92R11 while I'm imagining this `e2eDriver` will be used in a broader context, e.g. reporting timeline as in flutter/flutter#58789 , so I'm trying to make this function useful independent of the performance test context while still being useful in my project.

Reply:
As for by default no writing anything, `writeResponseData` will look like a duplicate. Maybe I should add an example to e2e as for demo the usage of this PR. If we decide to go that direction I'll open a new PR after this lands for that.